core/domain/feedback_services.py

```diff
@@ -209,12 +254,18 @@ def create_messages(
 
     # Get the threads at the outset, in order to check that there are models
     # corresponding to each of the thread_ids.
-    thread_models = feedback_models.GeneralFeedbackThreadModel.get_multi(
-        thread_ids)
+    thread_models_with_none = (
+        feedback_models.GeneralFeedbackThreadModel.get_multi(
+            thread_ids
+        )
+    )
+    thread_models: List[feedback_models.GeneralFeedbackThreadModel] = []
     thread_ids_that_do_not_have_models = []
-    for index, thread_model in enumerate(thread_models):
+    for index, thread_model in enumerate(thread_models_with_none):
         if thread_model is None:
             thread_ids_that_do_not_have_models.append(thread_ids[index])
+        else:
+            thread_models.append(thread_model)
```

Comment:
Hi @sagangwee,
Can you please take a look at this comment https://github.com/oppia/oppia/pull/15693#discussion_r939500140
Also, @seanlip sean please take a look at the comment that @sagangwee mentioned. 

Because both of you have a different opinion on the usage of assert, it would be very helpful for me if you guys come to a mutual decision ( like whether to use assert to not here. )

Reply:
I don't think Sean and I are actually conflicting here. Re Sean's comment:

> The assertion on its own is not sufficient here because someone may write another function that calls this one in the future. In general I think it is a good rule of thumb to use assertions only for conditions that have been already validated within the same function.

We should be able to assert not None below since we're explicitly checking no models are `None` on line 269.

Reply:
ohh really sorry!! I mistakenly misinterpreted it.
Thanks for pointing it out!

Reply:
I have tried, If we remove these new variables from here then we have to add 3 assertions ( at lines `310`, `335`, `348`) and 2 casts ( at lines `357`and `359`) because methods `update_timestamps_multi` and `put_multi` cannot accept Optional[...] values.

Reply:
I see. That does seem overly verbose in this case. SGTM.